### PR TITLE
fix: crashing heroku deployment

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link
n/a

## Implementation description
- [fix: Copy .cjs files to build/ directory after yarn build](https://github.com/uwblueprint/jump-math/pull/378/commits/2291ae1ba067a77a05ee1269402a3d9aea83af86)
- [fix: Remove the need for .cjs extensions for library files](https://github.com/uwblueprint/jump-math/pull/378/commits/be1d1a01cc303d6a30717f39054f83a75fedf9bc)
- [fix: Add eslint rule to catch manual imports of .cjs files](https://github.com/uwblueprint/jump-math/pull/378/commits/3ad3708f9156192c47226b75edcb67da8f2109da)
- [fix: Ignore build directory when using nodemon](https://github.com/uwblueprint/jump-math/pull/378/commits/9b54a78fda1eedca8a4bdc95a1d9032d688c2859)
- [fix: Lint ALL project files (not just .ts/.js files)](https://github.com/uwblueprint/jump-math/pull/378/commits/4bf4e5a9b80d09a879d500edfe15d6a26af6c1e6)

## Steps to test
0. Consider `rm -rf build/` for a fresh start.
1. Run `yarn build` locally to generate your `build/` directory.
2. Run `node build/server.js` and verify that server does not crash due to an import error (it might crash due to missing environment variables).

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by jfdoming** [comment]
Explanation: we're using the `graphql-upload` library for file uploads. However, the maintainer of that library made a decision (that I hate) to use CommonJS modules in their project in newer versions (and we updated to this newer version in order to fix a security vulnerability that actually had an impact unfortunately). CommonJS modules are like a virus (metaphorically) in that if you use a single CJS dependency, then normally you'd have to make your entire project CJS. Unfortunately, CJS is straight-up incompatible with a lot of other libraries we use, so that's not an option.

I solved this problem a while back by manually generating a few `.cjs` files and importing them, which works fine in development. However, the TypeScript compiler doesn't know about files with that extension by default, so at build time (which only occurs on Heroku), it was failing to copy those `.cjs` files to the `build/` directory, resulting in runtime errors. The solution is 1) to instruct TypeScript to copy all JavaScript files in the project into the `build/` directory via the `allowJs` option, and 2) to rename our `.cjs` files to `.js` files so that TypeScript will become aware of them (it doesn't know about the `.cjs` extension).

It's ugly, but it works, so...

**Review by joyce-shi — APPROVED** [review]
thank you for fixing! 😎 🔨 ( <- this is a picture of you)

**Comment by jfdoming** [comment]
lmao np. Unfortunately though this "fix" doesn't actually work in all cases (there are like 5 different things I need to check in order to make sure it works) 💀 working on a better fix, but let's not merge yet plz

also it makes me sad that our tests didn't catch any of this 😭

**Comment by jfdoming** [comment]
@joyce-shi the new fix is to manually copy all `.cjs` files from the `lib/` directory into `build/lib`. Not the best solution, but the best solution would be to migrate to an actual build tool like `babel` rather than using TypeScript directly, and that would be way more of a hassle.

File: backend/package.json
Comment (jfdoming) [inline backend/package.json:30]: This dependency was already being installed by other packages, so adding it here won't affect our usage at all!

**Review by joyce-shi — APPROVED** [review]

**Pull request merged**
